presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2331,11 +2346,19 @@ public void testGroupedExecution()
             assertQuery(colocatedAllGroupsAtOnce, joinThreeMixedTable, expectedJoinQuery);
             assertQuery(colocatedOneGroupAtATime, joinThreeBucketedTable, expectedJoinQuery);
             assertQuery(colocatedOneGroupAtATime, joinThreeMixedTable, expectedJoinQuery);
+            assertQuery(colocatedAllGroupsAtOnceDynamic, joinThreeBucketedTable, expectedJoinQuery);
+            assertQuery(colocatedAllGroupsAtOnceDynamic, joinThreeMixedTable, expectedJoinQuery);
+            assertQuery(colocatedOneGroupAtATimeDynamic, joinThreeBucketedTable, expectedJoinQuery);
+            assertQuery(colocatedOneGroupAtATimeDynamic, joinThreeMixedTable, expectedJoinQuery);
```

Comment:
I feel like it's excessive to run every tests with all the session property combinations.

`colocatedAllGroupsAtOnceDynamic` is probably not particularly interesting in general. As a result, it probably only needs to be tested for a small number of queries.

`colocatedOneGroupAtATimeDynamic` is probably not particularly interesting for queries with remote exchanges. As a result, this property should probably only be tested on a small number of such queries.